src/components/Trade/TradeRoutes/TradeRoutes.tsx

```diff
@@ -28,7 +28,7 @@ export const TradeRoutes = () => {
           render={(props: RouteComponentProps) => (
             <SelectAsset
               onClick={handleAssetClickWithAction(AssetClickAction.Sell)}
-              filterBy={getSupportedSellableAssets}
+              assets={supportedSellAssetsByMarketCap}
```

Comment:
Ditto those react-router are "just react" so they inherit from the same rules as any component - would it make sense to pass an action instead and select the right assets list from `<SelectAsset />` itself depending on the action?

As an added bonus, this would make `<TradeRoutes />` not use the `useSwapper()` hook anymore, which would probably avoid a lot of render cycles as well!

Reply:
yeah as above, we could pass an action instead of the literal list of assets, but in order to allow different assets to be rendered without computing inside the component we'd be drilling a selector instead of a list which seems similar unless im not understanding?